packages/serve/src/getDevServerOptions.ts

```diff
@@ -0,0 +1,24 @@
+/**
+ *
+ * Get the devServer option from the user's compiler options
+ *
+ * @param {Object} compiler - webpack compiler
+ *
+ * @returns {Object}
+ */
+export default function getDevServerOptions(compiler): any {
+    let devServerOptions;
+    if (compiler.compilers) {
+        // devServer options could be found in any of the compilers,
+        // so simply find the first instance and use it, if there is one
+        const comp = compiler.compilers.find((comp) => comp.options.devServer);
```

Comment:
We should let user decide which one to use. There are two possible solutions for this.
- Add an inquirer question with the names of compilers and let user select.
- Add a `--name` flag where user can supply the name of the compiler that should be used. (as suggested by @evilebottnawi)

Using both is also a solution (a better one imo).

Reply:
Think the `name` is better here, but in theory developers can run `two` dev servers, we need thing about it

Reply:
Yeah @evilebottnawi, I have also suggested to run all possible `devServer`s in parallel before, [here](https://github.com/webpack/webpack-cli/pull/1470#discussion_r443180108) is a small thread on the discussion.

Friendly ping @Loonride :)

Reply:
Will get back to this soon, there is still quite a few things to do on the dev server side first :smile:

Reply:
I agree with the `name` idea. Maybe it should also take a number index for the compiler being used, in case the user has not given their compilers names?

We don't currently have a way for the devServer to take multiple settings objects and run them all in parallel. Are you suggesting that we create this feature, or are you suggesting that we run multiple devServers on different ports and pass each individual devServer options in?

Reply:
> I agree with the name idea. Maybe it should also take a number index for the compiler being used, in case the user has not given their compilers names?

I think we should enforce `name` only for `name` flag instead of indices here as they are bound to change if we change order of compilers in subsequent calls. 

> Are you suggesting that we run multiple devServers on different ports and pass each individual devServer options in? I think it would make sense to throw an error if they don't specify unique ports for each config.

For a particular `devServer`, we assign it a port if it is there else look for a free port. Throwing errors in this case can be avoided leading to better UX.

P.S.
A possible issue (on CLI side) in this case would handling logging in CLI such that we know which compiler's devServer is logging on `stdout` of terminal, ( compiler `name` would be helpful here ).

Reply:
Ok, I switched `name` to not act as an index.

I'm just concerned with ports that if there is a lot of CLI output, it will be difficult for the user to even figure out which ports the server is running on when ports are assigned automatically

Reply:
> I'm just concerned with ports that if there is a lot of CLI output, it will be difficult for the user to even figure out which ports the server is running on when ports are assigned automatically

Yeah, this would be problem if `silent`/`quiet` is configured by a user (in other cases we can log this metadata to terminal).
Should we log this metadata (like port, name, etc) any way in that case as well 😅?

Regarding finding output of a particular compiler/port among a lot CLI output, we can configure logging using `name` of compiler as it's identifier and have output something like [concurrently](https://www.npmjs.com/package/concurrently).